examples/kelvin_helmholtz_instability.jl

```diff
@@ -0,0 +1,406 @@
+# This is not a requirement and can be disabled for speeding up the algorigithm but we'll keep it
+# here for illustration purposes.)
+"""
+    estimate_growth_rate!(simulation, energy, ω; convergence_criterion=1e-2)
+
+Estimates the growth rate iteratively until the relative change
+in the estimated growth rate ``σ`` falls below `convergence_criterion`.
+
+Returns ``σ``.
+"""
+function estimate_growth_rate!(simulation, energy, ω, b; convergence_criterion=1e-3)
+    σ = []
+    
+    plotseries = Array{Any, 1}(undef, 20)  # expect no more than 20 iterations
+    
+    compute!(ω)
+    plotseries[1] = powermethodplot(interior(ω)[:, 1, :], interior(b)[:, 1, :], σ, nothing)
+    
+    iteration = 1
+    while convergence(σ) > convergence_criterion
+        push!(σ, grow_instability!(simulation, energy))
+
+        compute!(energy)
+
+        @info @sprintf("Power method iteration %d, kinetic energy: %.2e, σⁿ: %.2e, relative Δσ: %.2e",
+                       length(σ), energy[1, 1, 1], σ[end], relative_difference(σ))
+
+        compute!(ω)
+        plotseries[iteration+1] = powermethodplot(interior(ω)[:, 1, :], interior(b)[:, 1, :], σ, nothing)
+        iteration += 1
+        
+        rescale!(simulation.model, energy)
+        compute!(energy)
+        
+        @info @sprintf("Kinetic energy after rescaling: %.2e", energy[1, 1, 1])
+    end
+
+    return σ, plotseries
+end
+nothing # hide
+
+# # Eigenplotting
+#
+# A good algorithm wouldn't be complete without a good visualization,
+
+using Oceananigans.AbstractOperations
+
+u, v, w = model.velocities
+b = model.tracers.b
+
+perturbation_vorticity = ComputedField(∂z(u) - ∂x(w))
+
+x, y, z = nodes(perturbation_vorticity)
+xb, yb, zb = nodes(b)
+
+eigentitle(σ, t) = length(σ) > 0 ? @sprintf("Iteration #%i; growth rate %.2e", length(σ), σ[end]) : @sprintf("Initial perturbation fields")
+eigentitle(::Nothing, t) = @sprintf("Vorticity at t = %.2f", t)
+
+function eigenplot(ω, b, σ, t; ω_lim=maximum(abs, ω)+1e-16, b_lim=maximum(abs, b)+1e-16)
+    kwargs = (xlabel="x", ylabel="z", linewidth=0, label=nothing, color = :balance, aspectratio = 1,)
+    plot_ω = contourf(x, z, clamp.(ω, -ω_lim, ω_lim)';
+                      levels = range(-ω_lim, stop=ω_lim, length=20),
+                      xlims = (grid.xF[1], grid.xF[grid.Nx]),
+                      ylims = (grid.zF[1], grid.zF[grid.Nz]),
+                      clims = (-ω_lim, ω_lim),
+                      title = "vorticity", kwargs...)
+    plot_b = contourf(xb, zb, clamp.(b, -b_lim, b_lim)';
+                    levels = range(-b_lim, stop=b_lim, length=20),
+                    xlims = (grid.xC[1], grid.xC[grid.Nx]),
+                    ylims = (grid.zC[1], grid.zC[grid.Nz]),
+                    clims = (-b_lim, b_lim),
+                    title = "buoyancy", kwargs...)
+    return plot(plot_ω, plot_b, layout=(1, 2), size=(800, 380))
+end
+
+function powermethodplot(ω, b, σ, t)
```

Comment:
Is `t=nothing` always here? It may be better to omit it from the signature.

Reply:
It's not nothing for the nonlinear equilibration.